service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteAzureContextFlightTest.java

```diff
@@ -0,0 +1,356 @@
+  // BaseConnectedTest which doesn't have azure enabled so it lives here for now. If/when test
+  // structure is re-evaluated and BaseAzureTest goes away, this test should be moved
+  @Test
+  void deleteMcWorkspaceWithAzureContextAndResource() throws Exception {
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+
+    // create new workspace so delete at end of test won't interfere with @AfterEach teardown
+    UUID uuid = UUID.randomUUID();
+    Workspace request =
+        Workspace.builder()
+            .workspaceId(uuid)
+            .userFacingId("a" + uuid.toString())
+            .workspaceStage(WorkspaceStage.MC_WORKSPACE)
+            .spendProfileId(spendUtils.defaultSpendId())
+            .build();
+    UUID mcWorkspaceUuid = workspaceService.createWorkspace(request, userRequest);
+
+    // Create the azure context.
+    FlightMap createParameters =
+        azureTestUtils.createAzureContextInputParameters(mcWorkspaceUuid, userRequest);
+    FlightState flightState =
+        StairwayTestUtils.blockUntilFlightCompletes(
+            jobService.getStairway(),
+            CreateAzureContextFlight.class,
+            createParameters,
+            CREATION_FLIGHT_TIMEOUT,
+            null);
+    assertEquals(FlightStatus.SUCCESS, flightState.getFlightStatus());
+    AzureCloudContext azureCloudContext =
+        workspaceService.getAuthorizedAzureCloudContext(mcWorkspaceUuid, userRequest).orElse(null);
+    assertNotNull(azureCloudContext);
+
+    // Create Azure IP resource
+    ApiAzureIpCreationParameters ipCreationParameters =
+        ControlledResourceFixtures.getAzureIpCreationParameters();
+
+    final UUID ipId = UUID.randomUUID();
+    ControlledAzureIpResource ipResource =
+        ControlledAzureIpResource.builder()
+            .common(
+                ControlledResourceFields.builder()
+                    .workspaceUuid(mcWorkspaceUuid)
+                    .resourceId(ipId)
+                    .name("wsm-test" + ipId.toString())
+                    .cloningInstructions(CloningInstructions.COPY_RESOURCE)
+                    .accessScope(AccessScopeType.fromApi(ApiAccessScope.SHARED_ACCESS))
+                    .managedBy(ManagedByType.fromApi(ApiManagedBy.USER))
+                    .build())
+            .ipName(ipCreationParameters.getName())
+            .region(ipCreationParameters.getRegion())
+            .build();
+
+    // Submit an IP creation flight.
+    flightState =
+        StairwayTestUtils.blockUntilFlightCompletes(
+            jobService.getStairway(),
+            CreateControlledResourceFlight.class,
+            azureTestUtils.createControlledResourceInputParameters(
+                mcWorkspaceUuid, userRequest, ipResource),
+            CREATION_FLIGHT_TIMEOUT,
+            null);
+    assertEquals(FlightStatus.SUCCESS, flightState.getFlightStatus());
+
+    // Run the delete flight, retrying every step once
```

Comment:
Not ever step is retried-- `DeleteControlledAzureResourcesStep.class.getName()` specifically is not.